mm/pmm.c

```diff
@@ -310,39 +300,36 @@ static size_t process_memory_range(unsigned index, unsigned first_avail_region)
      * because initial virtual memory mapping is small.
      */
 
-    /* Add initial 4K frames and align to 2M. */
-    while ((cur < MB(EARLY_VIRT_MEM) || cur % PAGE_SIZE_2M) && cur + PAGE_SIZE <= end) {
-        if (index <= first_avail_region)
```

Comment:
I think as long as the early frames are added within the physical memory covered by boot pagetable's identity mapping, this should be fine. And right now I think it should indeed be.